src/index.js

```diff
@@ -1,53 +1,189 @@
+  const localData = localStorage.getItem('items');
+  const parsedData = JSON.parse(localData);
+  const eachItem = document.querySelectorAll('.eachItem');
+  for (let i = 0; i < eachItem.length; i += 1) {
+    if (eachItem[i].classList.contains('strike')) {
+      parsedData[i].completed = true;
+    } else {
+      parsedData[i].completed = false;
+    }
+    localStorage.setItem('items', JSON.stringify(parsedData));
+  }
+};
+
+// Edit items
+const editItems = (oldItem) => {
+  const newInput = document.createElement('input');
+  newInput.type = 'text';
+  newInput.classList.add('new-input');
+  newInput.value = oldItem.textContent;
+  oldItem.replaceWith(newInput);
+  const local = localStorage.getItem('items');
+  const data = JSON.parse(local);
+  newInput.addEventListener('keypress', (e) => {
+    const newId = newInput.parentElement.id;
+    if (e.key === 'Enter') {
+      const oldItem = document.createElement('div');
+      oldItem.textContent = newInput.value;
+      newInput.replaceWith(oldItem);
+      const editedContainers = document.querySelectorAll('.new-item');
+      for (let i = 0; i < editedContainers.length; i += 1) {
+        if ((editedContainers[i].id) === newId) {
+          data[i].description = newInput.value;
+          localStorage.setItem('items', JSON.stringify(data));
+        }
+      }
+    }
+  });
+};
+
+// Remove items
+const removeItems = (li) => {
+  listOfItems.removeChild(li);
+  let count = 0;
+  const parsedItems = localStorage.getItem('items');
+  let localData = JSON.parse(parsedItems);
+  // Filter true elements
+  localData = localData.filter((item) => item.completed === false);
+  // Update index of elements
+  localData.map((item) => {
+    item.index = count;
+    count += 1;
+    return null;
+  });
+  localStorage.setItem('items', JSON.stringify(localData));
+};
+
+// Add new item
+const addNewItem = (newDescription) => {
+  const newItem = document.createElement('div');
+  newItem.classList.add('list');
+  newItem.innerHTML += `
+<input type="checkbox" class="checkbox">
+<span>${newDescription}</span>
+<span class="ellipsis">
+<i class="fa-solid fa-ellipsis-vertical"></i>
+</span>
+<span class="trashcan">
+<i class="fa-solid fa-trash-can"></i>
+</span>
 `;
+  listOfItems.appendChild(newItem);
+  const checkbox = document.querySelectorAll('.checkbox');
+  checkbox.forEach((checkboxInput) => {
+    checkboxInput.addEventListener('click', () => {
+      checkboxInput.nextElementSibling.classList.toggle('strike');
+      checkboxInput.parentNode.classList.toggle('clicked-on');
+      const ellipsisIcon = checkboxInput.parentNode.childNodes[5];
+      ellipsisIcon.classList.toggle('inactive-ellipsis');
+      const trashIcon = checkboxInput.parentNode.childNodes[7].childNodes[1];
+      trashIcon.classList.toggle('active-trash');
+      completeItem();
+    });
+  });
+  // Create and send new item to local storage
+  const newListItem = new ListItem(newDescription, false, checkbox.length - 1);
```

Comment:
- [ ] According to the project requirement, when you want to add a new task to the array, its `index` key should have a value that is equal to the length of the array + 1. This means that when you are creating the first task, the `index` of the first task should be 1 and not zero. You can simply set the index to `checkbox.length` instead of `checkbox.length - 1`. Kindly implement this. 👍